UI/src/view/dynamicAssetForm.js

```diff
@@ -0,0 +1,1010 @@
+          return '<option value="' + hour + '" ' + selected + '>' + hour + '</option>';
+        }).join('');
+      }
+
+      function minutesOptions(selectedOption) {
+        var range = _.range(0, 60, 5);
+        return _.map(range, function (minute) {
+          var selected = minute === selectedOption ? 'selected' : '';
+          return '<option value="' + minute + '" ' + selected + '>' + (minute<10 ? '0' + minute : minute) + '</option>';
+        }).join('');
+      }
+
+      var template = _.template('' +
+        '<div class="validity-period-group">' +
+        '<label>Voimassaoloaika (lisäkilvessä):</label>' +
+        ' <ul>' +
+        '   <%= existingValidityPeriodElements %>' +
+        newValidityPeriodElement() +
+        ' </ul>');
+
+      me.element = $(template({existingValidityPeriodElements: existingValidityPeriodElements}));
+
+      me.getPropertyValue = function(){
+        var values = me.getValue();
+        return me.createPropertyValue(values);
+      };
+
+      me.compare = function(propertyValueA, propertyValueB){
+        var isEqual = function(valueA , valueB) {
+          return valueA.startHour === valueB.startHour && valueA.startMinute === valueB.startMinute && valueA.endHour === valueB.endHour && valueA.endMinute === valueB.endMinute && valueA.days === valueB.days;
+        };
+
+        var isRemoved = function(firstProperty, secondProperty) {
+          _.remove(firstProperty.values, function (valuesA) {
+            return !_.isUndefined(_.find(secondProperty.values, function (valuesB) {
+              return isEqual(valuesB.value, valuesA.value);
+            }));
+          });
+          return _.isEmpty(firstProperty.values);
+        };
+        return isRemoved(_.cloneDeep(propertyValueA), _.cloneDeep(propertyValueB)) && isRemoved(_.cloneDeep(propertyValueB), _.cloneDeep(propertyValueA));
+      };
+
+      me.getValue = function() {
+        var periodElements = me.element.find('.existing-validity-period');
+        return _.map(periodElements, function (element) {
+          return { value: {
+            startHour: parseInt($(element).find('.start-hour').val(), 10),
+            startMinute: parseInt($(element).find('.start-minute').val(), 10),
+            endHour: parseInt($(element).find('.end-hour').val(), 10),
+            endMinute: parseInt($(element).find('.end-minute').val(), 10),
+            days: parseInt($(element).data('days'), 10)
+          }};
+        });
+      };
+
+      me.element.on('click', '.existing-validity-period .delete', function(event) {
+        $(event.target).parent().parent().remove();
+        me.setSelectedValue(setValue, getValue);
+      });
+
+      me.element.on('change', '.existing-validity-period .select', function(event) {
+        me.setSelectedValue(setValue, getValue);
+      });
+
+      me.element.on('change', '.new-validity-period select', function(event) {
+        disabled = "";
```

Comment:
is this 'disabled' declared somewhere else?

Reply:
No, it was replaced by me.disabled()